src/GafferOSL/OSLImage.cpp

```diff
@@ -369,3 +380,94 @@ IECore::ConstCompoundDataPtr OSLImage::computeShading( const Gaffer::Context *co
 
 	return result;
 }
+
+void OSLImage::updateChannels()
+{
+	// Disable undo for the actions we perform, because anything that can
+	// trigger an update is undoable itself, and we will take care of everything as a whole
+	// when we are undone.
+	UndoScope undoDisabler( scriptNode(), UndoScope::Disabled );
+
+	// Currently the OSLCode node will recompile every time an input is added.
+	// We're hoping in the future to avoid doing this until the network is actually needed,
+	// but in the meantime, we can save some time by emptying the code first, so that at least
+	// all the redundant recompiles are of shorter code.
+	oslCode()->codePlug()->setValue( "" );
+
+	oslCode()->parametersPlug()->clearChildren();
+
+	std::string code = "closure color out = 0;\n";
```

Comment:
Why don't we build into `Ci` directly, rather than using this intermediate `out` variable?